coalib/settings/Setting.py

```diff
@@ -129,12 +146,30 @@ def __path__(self, origin=None):
 
         if hasattr(self, "origin") and self.origin != "":
             origin = self.origin
+        if globbed_origin:
+            origin = glob_escape(origin)
 
         if origin is None:
             raise ValueError("Cannot determine path without origin.")
 
         return os.path.abspath(os.path.join(os.path.dirname(origin), strrep))
 
+    def __glob__(self, origin=None):
+        """
+        Determines the path of this setting with proper escaping of its
+        parent directories.
+
+        :param origin:      The origin file to take if no origin is specified
+                            for the given setting. If you want to provide a
+                            directory, make sure it ends with a directory
+                            seperator.
+        :return:            An absolute path in which the parent directories
+                            are escaped.
+        :raises ValueError: If no origin is specified in the setting nor the
+                            given origin parameter.
+        """
+        return Setting.__path__(self, origin, True)
```

Comment:
Maybe it's a better choice to do `Setting.__path__(self, glob_escape(origin))`, then you are able to remove the if condition above?